Develop

AlexanderTuzhikov opened the pull request.

AlexanderTuzhikov committed:
feat: реализована система рекомендаций фильмов

Добавлен метод findRecommendationsFilms для получения фильмов,
которые понравились пользователям с похожими вкусами.
Реализован SQL-запрос для нахождения пользователей с максимальным
пересечением лайков и формирования списка фильмов,
которые текущий пользователь ещё не оценивал.

Добавлен эндпоинт GET /users/{userId}/recommendations.

verp205 committed:
Сделал функционал «Общие фильмы» + исправил для тестов вывод mpa и genre

verp205 committed:
Сделал функционал «Общие фильмы» + исправил для тестов вывод mpa и genre

Ivan-Tyj committed:
Test version

AlexanderTuzhikov committed:
feat: реализована лента событий, ожидает реализации отзывов

AlexanderTuzhikov committed:
feat: реализован функционал удаления фильмов и пользователей

AlexanderTuzhikov committed:
fix: checkstyle

verp205 committed:
Добавил функционал отзывов + подправил application.yaml + добавил общий обработчик NotFoundException

verp205 committed:
Добавил в ReviewDbRepository @Language("SQL")

verp205 committed:
fix: CodeStyle

AlexanderTuzhikov committed:
Merge pull request #16 from AlexanderTuzhikov/add-remove-endpoint

feat: реализован функционал удаления фильмов и пользователей

AlexanderTuzhikov committed:
Merge branch 'develop' into add-recommendations

AlexanderTuzhikov committed:
Merge pull request #13 from AlexanderTuzhikov/add-recommendations

feat: реализована система рекомендаций фильмов

AlexanderTuzhikov committed:
Merge branch 'develop' into add-common-films

AlexanderTuzhikov committed:
Merge pull request #14 from AlexanderTuzhikov/add-common-films

Сделал функционал «Общие фильмы» + исправил для тестов вывод mpa и genre

AlexanderTuzhikov committed:
Merge remote-tracking branch 'origin/develop' into develop

AlexanderTuzhikov committed:
merge: add-recommendations, add-common-films, add-remove-endpoint

Ivan-Tyj committed:
Complete version: add directors

Ivan-Tyj committed:
Complete version: changed metods in FilmDbRepository

verp205 committed:
fix: Добавил наследование BaseDbRepositoryImpl + подправил методы реализации логики

AlexanderTuzhikov committed:
Merge branch 'develop' into add-reviews

AlexanderTuzhikov committed:
Merge pull request #17 from AlexanderTuzhikov/add-reviews

Добавил функционал отзывов + подправил application.yaml + добавил общ…

AlexanderTuzhikov committed:
Merge branch 'develop' into add-director-and-delete

Ivan-Tyj committed:
Complete version: add directors

Ivan-Tyj committed:
Complete version: checkstyle complete

Ivan-Tyj committed:
Complete version: checkstyle complete

Ivan-Tyj committed:
Complete version: checkstyle complete

Ivan-Tyj committed:
Complete version: jUnit tests fixed

AlexanderTuzhikov committed:
Merge branch 'develop' into add-feed

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/resources/schema.sql

AlexanderTuzhikov committed:
add-feed: реализована лента событий пользователя

Описание:
Добавлена возможность просмотра последних событий на платформе —
добавление в друзья, удаление из друзей, лайки и отзывы, которые оставили друзья пользователя.

- Создана таблица feed_events для хранения информации о действиях пользователей.
- Реализован API-эндпоинт GET /users/{id}/feed, возвращающий ленту событий пользователя.
- Включены события: FRIEND (ADD/REMOVE), LIKE (ADD/REMOVE), REVIEW (ADD/UPDATE/REMOVE).

Ivan-Tyj committed:
Complete version: postman tests fixed

AlexanderTuzhikov committed:
Merge pull request #21 from AlexanderTuzhikov/add-feed

Add feed

AlexanderTuzhikov committed:
Merge branch 'develop' into add-director

AlexanderTuzhikov committed:
Слить pull request #24

Add director

A-Glebov committed:
Добавили возможность выводить топ-N фильмов по количеству лайков с возможностью фильтрации по жанру и году релиза

A-Glebov committed:
Добавили возможность выводить топ-N фильмов по количеству лайков с возможностью фильтрации по жанру и году релиза

A-Glebov committed:
Merge branch 'develop' into add-most-populars

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

A-Glebov committed:
Добавили возможность выводить топ-N фильмов по количеству лайков с возможностью фильтрации по жанру и году релиза

AlexanderTuzhikov committed:
Merge pull request #27 from AlexanderTuzhikov/add-most-populars

Add most populars

Yulia772 committed:
Реализован поиск фильмов по ТЗ

AlexanderTuzhikov committed:
Merge branch 'develop' into add-search

AlexanderTuzhikov committed:
Слить pull request #28

Реализован поиск фильмов по ТЗ

AlexanderTuzhikov committed:
fix: postmantest

Yulia772 committed:
Исправлена реализация для тестов

verp205 committed:
доработал бд + подправил README файл

Yulia772 committed:
Fix logging and updateFilmDirectors logic

AlexanderTuzhikov committed:
test: 9 tests failed

verp205 committed:
Merge remote-tracking branch 'origin/develop' into develop

AlexanderTuzhikov committed:
Слить pull request #30

develop+

AlexanderTuzhikov committed:
test: 0 tests failed

AlexanderTuzhikov committed:
refactor: добавлен сервисный слой feed и event, убраны лишние методы из UserController, EventDbRepository

AlexanderTuzhikov committed:
refactor: добавлен сервисный слой FriendshipService и LikeService, отдал на них работу с лайками и дружбой чтобы разгрузить UserService и FilmService

AlexanderTuzhikov committed:
complete version 1.0

maks-avilow reviewed (changes requested):
Привет! Действительно хорошая работа, многие моменты мне понравились. Виден большой рост. Тем не менее, я оставил 4 комментария с подробным пояснением доработок. 2 комментария из них критичны и очень прошу их доработать. Они касаются использования энтити в контроллерах и проблемы n+1.  Еще 2 комментария(касаются логирования в контроллерах и ручного маппинга) остаются по вашему желанию. В любом случае это будет точка роста как разработчика. 
Жду ваших доработок!

AlexanderTuzhikov committed:
fix: DirectorController - отправляется только DTO на входе принимаются NewDirectorRequest и UpdateDirectorRequest
refactor: все логирование из controllers перенесено в service. Доработан DirectorService

Yulia772 committed:
Изменена реализация для решения проблемы n+1

AlexanderTuzhikov committed:
Слить pull request #32

Fix develop

AlexanderTuzhikov committed:
refactoring: FilmDbRepository - остались только CRUD
Ответственность по Search вынесена на FilmDbSearcher
Отдельно вынесен FilmRelationLoader отвечающий за загрузку фильмов со всеми полями
Работа с жанрами отдана в GenreDbRepository

Yulia772 committed:
changed method

Yulia772 committed:
Оптимизация FilmService для уменьшения запросов к базе

AlexanderTuzhikov committed:
refactoring: добавлено использование mapstruct

AlexanderTuzhikov committed:
Merge remote-tracking branch 'origin/fix-develop' into develop

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

AlexanderTuzhikov committed:
complete version 1.1

maks-avilow reviewed (approved):
Привет! вижу доработки, спасибо
Зачёт
